internal/clients/embed.go

```diff
@@ -0,0 +1,54 @@
+package clients
+
+import (
+	"context"
+	"embed"
+	"errors"
+	"strings"
+
+	"gopkg.in/yaml.v2"
+)
+
+var AwsSupported = make(map[string]interface{})
+var ErrArchitectureNotSupported = errors.New("architecture is not supported")
+
+const x86_64 = "x86_64"
+const arm64 = "arm"
+const i386 = "i386"
+
+type supportedInstanceTypes struct {
+	Aws struct {
```

Comment:
The root name should be something generic:

```suggestion
	SupportedTypes struct {
```

Reply:
I think it is double nesting, the whole struct is called supportedInstanceTypes / supportedTypes

Reply:
So it's always a good practice to have some kind of root element in XML/JSON/YAML, because later when you need to add just one more thing, you don't need to break it apart. So you can do:

```yaml
original_data:
  - xxx
  - xxx
some_additional_stuff_5_years_later:
  - xxx
  - xxx
  - xxx
```

But I am heavily biased by my years of XML programming where it was a problem, maybe YAML does not suffer from this. Then just remove the extra level I guess.